File: 2114-maximum-number-of-words-found-in-sentences/2114-maximum-number-of-words-found-in-sentences.cpp
class Solution {
public:
    int mostWordsFound(vector<string>& sentences) {
        int maxWord =0;
        for(string s: sentences){
            int cnt = 1; //to count the spaces in a sentence
            for(char c : s){
                if(c == ' ' ){
                    cnt++;
                }
            }
             maxWord = max(maxWord,cnt);
        }
        return maxWord;
    }
    
};

File: 2114-maximum-number-of-words-found-in-sentences/2114-maximum-number-of-words-found-in-sentences.java
class Solution {
    public int mostWordsFound(String[] sentences) {
        int maxWords = 0;
        for(String s : sentences){
            int count =1; //to count the space
            for(char ch : s.toCharArray()){
                if(ch == ' ')
                count++;
            }
            maxWords = Math.max(maxWords,count);
        }
        return maxWords;
    }
}